### PR TITLE
Modbus fixes

### PR DESCRIPTION
**Pull request by shahardo**

### CODE REVIEW

File: negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php [hunk 1]
```diff
@@ -51,10 +51,53 @@ public function publicFieldsInfo() {
...
+   * Return the basic entity field query for messages, with additional filter
+   * that matches only messages accessible by the current user.
+   */
+  public function getEntityFieldQuery() {
+
+    $query = parent::getEntityFieldQuery();
+
+    // Add condition to match only messages accessible by the current user.
+    // Find the list of valid OGs for current user
+    global $user;
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php]: `$account = $this->getAccount();`

File: negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php [hunk 2]
```diff
@@ -51,10 +51,53 @@ public function publicFieldsInfo() {
...
+    if (empty($user->og_user_node)) {
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php]: `$wrapper = entity_metatda...`

File: negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php [hunk 3]
```diff
@@ -51,10 +51,53 @@ public function publicFieldsInfo() {
...
+      return $query;
+    }
+    $og_groups = array();
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/message/anomalous_consumption/1.0/NegawattMessageAnomalousConsumption.class.php]: `$gids = $wrapper->og_user_node->value(array('identifier' => TRUE));`

File: negawatt/negawatt.info [hunk 4]
```diff
@@ -34,6 +34,7 @@ dependencies[] = geolocation_googlemaps
 dependencies[] = libraries
 dependencies[] = logs_http
 dependencies[] = message
+dependencies[] = message_ui
```
Comment (amitaibu) [inline negawatt/negawatt.info]: remove - move to `./install` (same as enabling field ui)

**Comment by shahardo** [comment]
Merged

**Pull request closed without merging**